src/crypto/mod.rs

```diff
@@ -956,64 +934,65 @@ impl<'de> Deserialize<'de> for COSEKey {
                                     SerdeError::custom(format!("unsupported curve {value}"))
                                 })?;
                                 curve = Some(val);
-                                // curve = Some(map.next_value()?);
                             }
-                        }
-                        -2 => {
-                            if x.is_some() {
-                                return Err(SerdeError::duplicate_field("x"));
+                            Some(COSEKeyTypeId::RSA) => {
+                                if n.is_some() {
+                                    return Err(SerdeError::duplicate_field("n"));
+                                }
+                                let value: ByteBuf = map.next_value()?;
+                                n = Some(value.to_vec());
                             }
-                            let value: ByteBuf = map.next_value()?;
-                            x = Some(value.to_vec());
-                        }
-                        -3 => {
+                        },
+                        -2 => match key_type {
+                            None => return Err(SerdeError::missing_field("key_type")),
+                            Some(COSEKeyTypeId::OKP) | Some(COSEKeyTypeId::EC2) => {
+                                if x.is_some() {
+                                    return Err(SerdeError::duplicate_field("x"));
+                                }
+                                let value: ByteBuf = map.next_value()?;
+                                x = Some(value.to_vec());
+                            }
+                            Some(COSEKeyTypeId::RSA) => {
+                                if e.is_some() {
+                                    return Err(SerdeError::duplicate_field("e"));
+                                }
+                                let value: ByteBuf = map.next_value()?;
+                                e = Some(value.to_vec());
+                            }
+                        },
+                        -3 if key_type == Some(COSEKeyTypeId::EC2) => {
```

Comment:
If the key type isn't EC2 (or isn't present), do we just drop the value? (in other words, are superfluous entries in the map an issue?)

Reply:
If the key type isn't EC2, we'll move down to the `other` branch of the match statement and return an error.

Reply:
Ohhh, right - I was misreading this.